Command/CrawlCommand.php

```diff
@@ -105,7 +105,7 @@ public function crawlUrl(AMQPMessage $message)
     {
         $crawlJob = CrawlJob::create($message);
 
-        if (!$this->areHostsEqual($crawlJob->getUrl(), $crawlJob->getBaseUrl())) {
+        if (!$this->areHostsEqual($crawlJob->getUrl(), $crawlJob->getBaseUrl()) && !CrawlJob::isUrlWhitelisted($crawlJob->getUrl(), $crawlJob->getWhitelist())) {
```

Comment:
Incorrect check; because the whitelist might 'whitelist' domains that are not equal. So actually first the whitelist check should be done and then the hosts check (if no whitelist pattern matches).

I would also like to ask you to move this `areHostsEqual` method to the CrawlJob class. Then have a public method `$crawlJob->isWhitelisted()` (or maybe `isAllowedToCrawl` and include all the logic in there, what do you think?